Mathlib/RingTheory/LocalProperties/Submodule.lean

```diff
@@ -99,3 +106,182 @@ theorem Module.subsingleton_of_localization_maximal
   rw [subsingleton_iff_forall_eq 0]
   intro x
   exact Module.eq_of_localization_maximal Rₚ Mₚ f x 0 fun _ _ ↦ Subsingleton.elim _ _
+
+section localization_maximal
+
+open Submodule LocalizedModule Ideal LinearMap
+
+/-- Another version, using `LocalizedModule` instead of `IsLocalizedModule`. -/
+lemma Submodule.le_of_localization_maximal' {N H : Submodule R M}
+    (h : ∀ (J : Ideal R) [J.IsMaximal], localized J.primeCompl N ≤ localized J.primeCompl H) :
+    N ≤ H := Submodule.le_of_localization_maximal (fun P ↦ (Localization P.primeCompl))
+    (fun P ↦ (LocalizedModule P.primeCompl M))
+    (fun P ↦ (mkLinearMap P.primeCompl M)) (fun P _ ↦ h P)
+
+/-- Another version, using `LocalizedModule` instead of `IsLocalizedModule`. -/
+lemma Submodule.eq_of_localization_maximal' {N P : Submodule R M}
+    (h : ∀ (J : Ideal R) (_ : J.IsMaximal), localized J.primeCompl N = localized J.primeCompl P) :
+    N = P :=
+  eq_of_le_of_le (le_of_localization_maximal' (fun J hJ ↦ le_of_eq (h J hJ)))
+  (le_of_localization_maximal' (fun J hJ ↦ le_of_eq (h J hJ).symm))
+
+/-- Another version, using `LocalizedModule` instead of `IsLocalizedModule`. -/
+lemma Submodule.eq_bot_of_localization_maximal' {N : Submodule R M}
+    (h : ∀ (J : Ideal R) (_ : J.IsMaximal), localized J.primeCompl N = ⊥) :
+  N = ⊥ := eq_of_localization_maximal' fun _ _ ↦ by simp only [h, localized'_bot]
+
+/-- Another version, using `LocalizedModule` instead of `IsLocalizedModule`. -/
+lemma Submodule.eq_top_of_localization_maximal' {N : Submodule R M}
+    (h : ∀ (J : Ideal R) (_ : J.IsMaximal), localized J.primeCompl N = ⊤) :
+  N = ⊤ := eq_of_localization_maximal' fun _ _ ↦ by simp only [h, localized'_top]
+
+end localization_maximal
+
+section span
+
+open IsLocalizedModule LocalizedModule Ideal
+
+variable {R M : Type*} [CommRing R] [AddCommGroup M] [Module R M]
+  (s : Set R) (spn : Ideal.span s = ⊤)
+include spn
+
+theorem eq_zero_of_localization_span (x : M)
+    (h : ∀ r : s, (mkLinearMap (Submonoid.powers r.1) M ) x = 0) : x = 0 := by
```

Comment:
#19126 now contains `span` lemmas too, but only the `IsLocalizedModule` version. I've been able to golf the two largest proofs. Please feel free to copy from or merge #19126, preferably after #19126 is merged into master.